src/main/java/org/jenkinsci/plugins/gitclient/GitToolConfigurator.java

```diff
@@ -0,0 +1,127 @@
+package org.jenkinsci.plugins.gitclient;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.plugins.git.GitTool;
+import hudson.tools.ToolProperty;
+import io.jenkins.plugins.casc.Attribute;
+import io.jenkins.plugins.casc.BaseConfigurator;
+import io.jenkins.plugins.casc.ConfigurationContext;
+import io.jenkins.plugins.casc.Configurator;
+import io.jenkins.plugins.casc.ConfiguratorException;
+import io.jenkins.plugins.casc.model.CNode;
+import io.jenkins.plugins.casc.model.Mapping;
+import io.jenkins.plugins.casc.model.Sequence;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+@Extension(optional = true)
+public class GitToolConfigurator extends BaseConfigurator<GitTool> {
+
+    @NonNull
+    @Override
+    public String getName() {
+        return "git";
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Git";
+    }
+
+    @Override
+    public Class getTarget() {
+        return GitTool.class;
+    }
+
+    @Override
+    public boolean canConfigure(Class clazz) {
+        return clazz == GitTool.class;
+    }
+
+    @NonNull
+    @Override
+    public Class getImplementedAPI() {
+        return getTarget();
+    }
+
+    @NonNull
+    @Override
+    public List<Configurator<GitTool>> getConfigurators(ConfigurationContext context) {
+        return Collections.singletonList(this);
+    }
+
+    @NonNull
+    @Override
+    public List<Attribute<GitTool, ?>> getAttributes() {
+        Attribute<GitTool, String> name = new Attribute<>("name", String.class);
+        Attribute<GitTool, String> home = new Attribute<>("home", String.class);
+        Attribute<GitTool, ToolProperty> p = new Attribute<>("properties", ToolProperty.class);
+        p.multiple(true);
+        return Arrays.asList(name, home, p);
+    }
+
+    @Override
+    protected GitTool instance(Mapping mapping, ConfigurationContext context) throws ConfiguratorException {
+        final CNode mproperties = mapping.remove("properties");
+        final String name = mapping.getScalarValue("name");
+        if (JGitTool.MAGIC_EXENAME.equals(name)) {
+            mapping.remove("home"); //Ignored but could be added, so removing to not fail handleUnknown
```

Comment:
What does "ignored but could be added" mean? That it is not taken into account right now?

Reply:
My understanding is that `home` is a field for other GitTool objects but not for this type. In case the yaml file contains that field it's ignored so we don't get an error. If I'm wrong, please correct me @rsandell !

Reply:
Then I think we should raise an error? Otherwise people might think the configuration is actually used, when it's not.

Reply:
Since other GitTools are using the field, it might be expected by users to be used, so putting the field might a recurrent mistake that always would cause an error. On the other hand, it's true that is a bad configuration. @rsandell how is the resulting exported yaml file? Maybe it's easier to make the decision with this input.
Beside, I think the maintainer can shed some light on this decision. CC @MarkEWaite

Reply:
Maybe not a error, but a warning. Doesn't seems to break anything.

Reply:
Yes, home is hard coded in jgit and jgitapache, but if someone exported the configuration without this configurator, like an old version of git-tool then the `home` parameter would be there since it is there in all ToolInstallers. So I didn't want to break the config in that case.

Reply:
makes sense. What about a warning as @alecharp proposed?

Reply:
It makes sense not failing then, but I think a warning is not a bad idea.

Reply:
`BaseConfigurator` will run a check after creating the instance to see, among other things, if there is anything left. Since I allow `home` to be set in general but ignored for the two other types I need to remove any potential left overs here so that check doesn't fail. I was toying with the idea of adding a warning log or something. But since I got the `describe` method working I didn't.

Reply:
Warn where? The system log?

Reply:
I was thinking about the typical logger, like https://github.com/jenkinsci/git-client-plugin/blob/55b186b5ad6c362caeb27d077ff153a6263fba22/src/main/java/hudson/plugins/git/GitTool.java#L197

Reply:
I found a `context.warning` that nicely puts the warning into an administrative monitor. But unfortunately there are other listeners to that warning that interprets it as an error :disappointed:

Reply:
added a jul logger